mantidproject/mantid
# Remove unused ThresholdRange class from VatesAPI

Description of work.

This removes `ThresholdRange` and derived class from the VatesAPI. I *think* only `NoThresholdRange` and `IgnoreZerosThresholdRange` are being used and the min and max values they calculate are ignored. The `ThresholdRangeWidget` is not part of the VSI. 

This class slows down creation of a `vtkDataSet` by adding a virtual function call in a tight loop and making it considerably more difficult to parallelize that loop. 

**To test:**



1. Verify my logic above for removing `ThresholdRange`. Review changes verifying this doesn't significantly change the user experience.
2. Load a MDHistoWorkspace, open it in the VSI, change views and verify the image is unchanged. 
3. Load a MDEventWorkspace, open it in the VSI, change views and verify the image is unchanged.

There is no GitHub issue 

*Does not need to be in the release notes.* This is am implementation detail and mentioning it in the release notes is more likely to confuse rather than help users.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
